DirectXTex/DirectXTexDDS.cpp

```diff
@@ -321,7 +361,7 @@ namespace
 
         // Verify header to validate DDS file
         if (pHeader->size != sizeof(DDS_HEADER)
-            || pHeader->ddspf.size != sizeof(DDS_PIXELFORMAT))
+            || pHeader->ddspf.size != sizeof(DDS_PIXELFORMAT) && !IsUT2004PixelFormat(pHeader->ddspf))
```

Comment:
For this check rather than introduce the extra helper functions, I suggest just going with:

```
// Verify header to validate DDS file
if (pHeader->size != sizeof(DDS_HEADER))
{
    return E_FAIL;
}

if (pHeader->ddspf.size != 0 && pHeader->ddspf.size != sizeof(DDS_PIXELFORMAT))
{
    return E_FAIL;
}
```